tests/Doctrine/Tests/ORM/Functional/Ticket/GH6699Test.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Tests\Models\CMS\CmsUser;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @group #6699
+ */
+class GH6699Test extends OrmFunctionalTestCase
+{
+    protected function setUp()
+    {
+        $this->useModelSet('cms');
+        parent::setUp();
+    }
+
+    public function testMixedParametersWithZeroNumber()
+    {
+        $qb = $this->_em->createQueryBuilder()
```

Comment:
We can simplify this bit, since the query builder reference is not really needed, like:

```php
$query = $this->_em->createQueryBuilder()
                   ->select('u')
                   ->from(CmsUser::class, 'u')
                   ->andWhere('u.username = :username')
                   ->andWhere('u.id = ?0')
                   ->getQuery();
```